src/services/ioServerRunner/ioServerRunner.ts

```diff
@@ -0,0 +1,29 @@
+import { Server } from "socket.io";
+import { TIoServerRunnerArgs } from "./ioServerRunner.types";
+import { newSubscriberListener } from "../../datafeed/newSubscriberListener";
+import { providerListener } from "../../datafeed/providerListener";
+import { currentConfig } from "../../utils/appconstants";
+import { EEpochEmitterNames, TOpfProvidedPredictions, epochEmitters } from "../../metadata/config";
+
+export const ioServerRunner = ({ httpServer }: TIoServerRunnerArgs) => {
+  const io = new Server(httpServer, {
+    path: "/api/datafeed",
+  });
+
+  newSubscriberListener({
+    io,
+  });
+
+  Object.entries(epochEmitters).forEach(
+    ([key, epochEmitterName]: [keyof TOpfProvidedPredictions, EEpochEmitterNames]) => {
+      const contractAddresses = currentConfig.opfProvidedPredictions[key];
+      if (!contractAddresses) return;
+
+      providerListener({
```

Comment:
Looks like there is a provider listener for each contract from a timeframe, can't we just creat a listener for each timeframe?